### ISSUE TITLE
BatchNorm2d is not traced in the torch graph?

### ISSUE THREAD

**Issue by twmht** [microsoft/nni#3193]
**Environment**:
- NNI version:v2.0
- NNI mode (local|remote|pai):local

**What issue meet, what's expected?**:

I am compressing  vovnet (https://github.com/youngwanLEE/vovnet-detectron2/blob/master/vovnet/vovnet.py) with slim.

the mask file has been generated and the accuracy is good, but when speed up the model it would say the batchnorm2d layer is not traced in the graph.

**How to reproduce it?**: 

I tried to train vovnet with slim_torch_cifar10.py

and then use model_speedup.py to speed up the model.

**Additional information**:

```
[12/14/2020, 05:40:59 PM] INFO (nni.compression.torch.speedup.compressor) start to speed up the model
[12/14/2020, 05:40:59 PM] INFO (nni.compression.torch.speedup.compressor) fix the mask conflict of the interdependent layers
[12/14/2020, 05:41:00 PM] INFO (nni.compression.torch.utils.mask_conflict) {'stage2.OSA2_1.ese.fc': 1, 'stage3.OSA3_1.ese.fc': 1, 'stage4.OSA4_1.ese.fc': 1}
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.utils.mask_conflict) no multi-dimension masks found.
[12/14/2020, 05:41:00 PM] INFO (nni.compression.torch.utils.mask_conflict) detected conv prune dim: 0
[12/14/2020, 05:41:00 PM] INFO (nni.compression.torch.speedup.compressor) infer module masks...
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stem.stem_1/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stem.stem_2/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stem.stem_3/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage2.OSA2_1.layers.0.OSA2_1_0/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage2.OSA2_1.layers.1.OSA2_1_1/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage2.OSA2_1.layers.2.OSA2_1_2/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage2.OSA2_1.concat.OSA2_1_concat/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage3.OSA3_1.layers.0.OSA3_1_0/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage3.OSA3_1.layers.1.OSA3_1_1/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage3.OSA3_1.layers.2.OSA3_1_2/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage3.OSA3_1.concat.OSA3_1_concat/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage4.OSA4_1.layers.0.OSA4_1_0/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage4.OSA4_1.layers.1.OSA4_1_1/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage4.OSA4_1.layers.2.OSA4_1_2/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] WARNING (nni.compression.torch.speedup.compressor) stage4.OSA4_1.concat.OSA4_1_concat/norm has mask, but not found in the traced graph, just skip it.
[12/14/2020, 05:41:00 PM] INFO (nni.compression.torch.speedup.compressor) replace compressed modules...
[12/14/2020, 05:41:00 PM] INFO (nni.compression.torch.speedup.compressor) speedup done
```

**Comment by twmht** [comment]
I tried to print the traced node. batchnorm2d is traced in the graph actually. 

but the name is different from mask file

for example, the first three layer names from traced graph are 

```
dict_keys(['conv.aten::_convolution.0', 'norm.aten::batch_norm.1', 'relu.aten::relu.2']))
```

any idea?

**Comment by zheng-ningxin** [comment]
Thanks for the feedback. The BN layers you print and the BN layers in the warning are not the same layers.

This warning means that these BN layers are not used in your forward function, but we found you have assign masks to these layers. There are basically two reasons (1) These BN layers are redundant to the model (2) Your model contains value-dependent judgments, such as if-else, and the input only triggered one of the paths.

**Comment by twmht** [comment]
@zheng-ningxin 

I found out the root cause is  the use of `torch.nn.Sequential(OrderedDict(...))`, which may cause name mismatch.

after replacing with `torch.nn.Sequential` without `OrderedDict`, the mask name would match the graph node name, and thus batchnorm layers can be pruned.

I doubt this may be a bug with pytorch jit, not nni.

**Comment by twmht** [comment]
by the way, is there any tutorial or introduction about how nni prune these masked layers? thiis may help if users want to add new network or new layers when doing compression.

**Comment by twmht** [comment]
I found out another new  issue after running `model_speedup.py`, I will create a newthread for this new issue.

**Comment by twmht** [comment]
see #3200  for related issues.
